web/apps/labelstudio/src/pages/DataManager/DataManager.jsx

```diff
@@ -140,7 +140,20 @@ export const DataManagerPage = ({ ...props }) => {
           },
         });
 
-        lsf.loadSuggestions(suggestionsRequest, (response) => {
+        // we'll check that we are processing the same task
+        const wrappedRequest = new Promise(async (resolve, reject) => {
+          const response = await suggestionsRequest;
+
+          // right now task might be an old task,
+          // so in order to get a current one we need to get it from lsf
+          if (task.id === dataManager.lsf.task.id) {
+            resolve(response);
+          } else {
+            reject();
```

Comment:
if we `reject()` then we move on to `lsf.loadSuggestions(wrappedRequest, (response) => {` wouldn't response be `undefined`? if thats the case line 157 `if (response.data) {` will throw an error most likely

Reply:
ahh, disregard - we catch in loadSuggestions so we dont fire `dataParser`. Resolving convo